dbc2val/dbcfeederlib/dbc2vssmapper.py

```diff
@@ -0,0 +1,310 @@
+        self.last_time : float = 0.0
+        # For value comparison (on_changes) we store last value used for comparison
+        self.last_vss_value : Any = None
+
+    def time_condition_fulfilled(self, time : float) -> bool:
+        """
+        Checks if time condition to send signal is fulfilled
+        Value (on_change) condition not evaluated
+        """
+        fulfilled = True
+        log.debug(f"Checking interval for {self.vss_name}. "
+                  f"Time is {time}, last sent {self.last_time}")
+
+        # First shall always evaluate to true
+        if (self.interval_ms > 0) and (self.last_time != 0.0):
+            diff_ms =( time - self.last_time) * 1000.0
+            if diff_ms < self.interval_ms:
+                log.debug(f"Interval not exceeded for {self.vss_name}. Time is {time}")
+                fulfilled = False
+
+        # We must set time already now even if a value check is performed later
+        # Reason is that value condition is evaluated later after queuing
+        if fulfilled:
+            self.last_time = time
+
+        return fulfilled
+
+    def change_condition_fulfilled(self, vss_value : Any) -> bool:
+        """
+        Checks if change condition to send signal is fulfilled.
+        Transformation is expected to be costly, so transformation and value check only performed
+        if time condition is fulfilled.
+        """
+        fulfilled = False
+        log.debug(f"Checking change condition for {self.vss_name}. "
+                  f"New value {vss_value}, old value {self.last_vss_value}")
+
+        if not vss_value is None:
+            if self.last_vss_value is None:
+                # Always send first value
+                fulfilled = True
+            elif self.on_change:
+                fulfilled = self.last_vss_value != vss_value
+            else:
+                fulfilled = True
+
+        if fulfilled:
+            self.last_vss_value = vss_value
+        return fulfilled
+
+
+
+    def transform_value(self, value : Any) -> Any:
+        """
+        Transforms the given "raw" DBC value to the wanted VSS value.
+        For now does not make any type checks
+        """
+        vss_value = None
+        if self.transform is None:
+            log.debug(f"No mapping to VSS {self.vss_name}, using raw value {value}")
+            vss_value = value
+        else:
+            # It is supposed that "extract_verify_transform" already have checked that
+            # we have a valid transform, so no need to check it here
+            if "mapping" in self.transform:
+                tmp = self.transform["mapping"]
+                # Assumed to be a list
+                for item in tmp:
+                    from_val = item["from"]
+                    if from_val == value:
+                        new_val = item["to"]
+                        vss_value = new_val
+                        break
+            else:
```

Comment:
Would it be more clean/future proof to really check there is a math transform? And in else print something like "there seem to be transforms I just don't understand them"

I am thinking that might be more robust, if we add further transform types in the future and overlays/config files/dbcfeeder implementations get mixed

Reply:
The idea is that `extract_verify_transform` shall do the static check that mappings are understood, so here only well defined mappings shall exist. But we could add an else-path here anyway. As long as we do not have too smart linters (that complains that the code is dead) it shall be ok.